cmd/gen-stdlib-allowlists/scan.go

```
@@ -0,0 +1,611 @@
+
+// classifyEntry routes a walker entry to the appropriate short-circuit:
+// skip=true means the caller should return action immediately. A nil
+// action coupled with skip=true is a silent drop; filepath.SkipDir
+// prunes the subtree. Folding the filters here lets visitGorootEntry
+// stay under KTN-FUNC-CYCLO and KTN-VAR-MERGEGUARD; ordering action
+// first violates KTN-FUNC-ERRLAST when it is `error`, so we return
+// `skip` first.
+func classifyEntry(path string, d fs.DirEntry, walkErr error) (skip bool, action error) {
+	//: Walker errors, null entries, or non-directories cannot contain packages.
+	if walkErr != nil || d == nil || !d.IsDir() {
+		return true, nil //: Drop walker errors, nil entries, and non-directories without action.
```

golangci-lint (2.12.2)

[error] error is not nil (line 112) but it returns nil

(nilerr)